### PR TITLE
Backport .38 and .39 migrations to master

### PR DESCRIPTION
**Pull request by ggwpez**
Make it possible to skip deployment of .42 by back-porting .38 and .39. Note that .42 contains those already.  
https://github.com/paritytech/substrate/pull/13417 should should ensure that we have all migrations.

### CODE REVIEW

File: runtime/kusama/src/lib.rs [hunk 1]
```diff
@@ -1461,14 +1461,29 @@ impl Get<Perbill> for NominationPoolsMigrationV4OldPallet {
...
 
+	pub type V0938 = (
+		// "Use 2D weights in XCM v3" <https://github.com/paritytech/polkadot/pull/6134>
+		pallet_xcm::migration::v1::MigrateToV1<Runtime>,
+		parachains_ump::migration::v1::MigrateToV1<Runtime>,
+		// Remove stale entries in the set id -> session index storage map
+		// (after this release they will be properly pruned after the bonding duration has elapsed)
+		pallet_grandpa::migrations::CleanupSetIdSessionMap<Runtime>,
+	);
+	pub type V0939 = ();
```
Comment (bkchr) [inline runtime/kusama/src/lib.rs:1486]:
```suggestion
	pub type V0939 = ();
```

Kusama is already on `0.9.38`

File: runtime/kusama/src/lib.rs [hunk 2]
```diff
@@ -1461,14 +1461,29 @@ impl Get<Perbill> for NominationPoolsMigrationV4OldPallet {
 ///
 /// This contains the combined migrations of the last 10 releases. It allows to skip runtime
 /// upgrades in case governance decides to do so. THE ORDER IS IMPORTANT.
-pub type Migrations =
-	(migrations::V0940, migrations::V0941, migrations::V0942, migrations::Unreleased);
+pub type Migrations = (
+	migrations::V0938,
```
Comment (bkchr) [inline runtime/kusama/src/lib.rs:1465]:
```suggestion
```

File: runtime/rococo/src/lib.rs [hunk 3]
```diff
@@ -1478,14 +1478,29 @@ pub type UncheckedExtrinsic =
...
+	migrations::V0942,
+	migrations::Unreleased,
+);
 
 /// The runtime migrations per release.
 #[allow(deprecated, missing_docs)]
 pub mod migrations {
 	use super::*;
 
+	pub type V0938 = (
```
Comment (bkchr) [inline runtime/rococo/src/lib.rs:1495]: Probably the same as for Kusama?

**Review by bkchr — APPROVED** [review]

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot/-/jobs/2811803

**Comment by ggwpez** [comment]
Migrations for Polkadot, Kusama and Westend turn green together with https://github.com/paritytech/polkadot/pull/7199

**Review by sam0x17 — APPROVED** [review]

**Comment by ggwpez** [comment]
done already

**Pull request closed without merging**